### PR TITLE
Upgrade to WasmVM v0.15.1

### LINKED ISSUE #532: Upgrade wasmvm to 0.15.0
Not yet released. Will follow shortly after https://github.com/CosmWasm/cosmwasm/milestone/17 is done.

### PR DESCRIPTION
**Pull request by alpe**
Closes #532

- [x] Messages and submessages are merged; the `replyOn` type now got a new "never" case for the fire and forget messages
- [x] `replyOn` is not an integer but supports the Stringer interface (i.e. `.String()`)
- [x] The interface version was bumped from interface_version_5 to interface_version_6 such that CosmWasm 0.15.0 contracts need to be used
- [x] In some Wasm messages "send" was renamed to "funds" for consistency
- [x] The IBC acknowledgement type was changed slightly
- [x] There is a new Gov message (MsgVote) that needs to be connected
- [x] Deserializing contract results now cost gas. This needs to be configured in a new argument of `vm.Execute` and friends. We messed up storing the value in the `vm` instance, such that it needs to be passed to every call. Sorry for that! The deserialization cost is in gas per byte. It is a fraction, so you can use 20/1, 10/1, 1/1, 1/2, 1/30, ... in there.
- [x] Contract responses now contain events. The exiting `Attributes` should go into a main `wasm` event as before. All other events should be added with a `wasm-` prefix on the event type.
- [x] Set correct release version in `go.mod`
- [x] Upgrade Dockerfile with proper version checksum

## Blocker
~~https://github.com/CosmWasm/wasmvm/pull/235~~
~~https://github.com/CosmWasm/wasmvm/issues/236~~

## Inconsistent
https://github.com/CosmWasm/wasmd/issues/552
https://github.com/CosmWasm/wasmd/issues/553

## Improvements:
https://github.com/CosmWasm/wasmvm/issues/239
~~https://github.com/CosmWasm/wasmvm/issues/238~~

### CODE REVIEW

**Comment by webmaster128** [comment]
A patched wasmvm 0.15.1 is released that should solve the two blocker tickets as well as the missing pointer in GovMsg.

File: x/wasm/relay_pingpong_test.go [hunk 1]
```diff
@@ -80,43 +79,19 @@ func TestPinPong(t *testing.T) {
...
-
-		seq := uint64(i)
-		pkg := channeltypes.NewPacket(ball.GetBytes(), seq, channelA.PortID, channelA.ID, channelB.PortID, channelB.ID, doNotTimeout, 0)
-		ack := ball.BuildAck()
-
-		err = coordinator.RelayPacket(chainA, chainB, clientA, clientB, pkg, ack.GetBytes())
-		require.NoError(t, err)
-		//coordinator.CommitBlock(chainA, chainB)
-		err = coordinator.UpdateClient(chainA, chainB, clientA, ibcexported.Tendermint)
+		err := coordinator.RelayAndAckPendingPackets(chainA, chainB, clientA, clientB)
```
Comment (ethanfrey) [inline x/wasm/relay_pingpong_test.go:89]: Did that just get 90% smaller? 😄

File: x/wasm/types/types.go [hunk 2]
```diff
@@ -297,23 +297,6 @@ func NewWasmCoins(cosmosCoins sdk.Coins) (wasmCoins []wasmvmtypes.Coin) {
 	return wasmCoins
 }
 
-// ParseEvents converts wasm LogAttributes into an sdk.Events. Returns events and number of bytes for custom attributes
-func ParseEvents(wasmOutputAttrs []wasmvmtypes.EventAttribute, contractAddr sdk.AccAddress) sdk.Events {
```
Comment (ethanfrey) [inline x/wasm/types/types.go:301]: Yup, I prefer your more complex functions now

File: x/wasm/types/wasmer_engine.go [hunk 3]
```diff
@@ -134,80 +140,87 @@ type WasmerEngine interface {
...
 		gasLimit uint64,
+		deserCost wasmvmtypes.UFraction,
 	) (*wasmvmtypes.IBCBasicResponse, uint64, error)
 
 	// IBCPacketTimeout is available on IBC-enabled contracts and is called when an
 	// outgoing packet (previously sent by this contract) will provably never be executed.
 	// Usually handled like ack returning an error
 	IBCPacketTimeout(
-		codeID wasmvm.Checksum,
+		checksum wasmvm.Checksum,
```
Comment (ethanfrey) [inline x/wasm/types/wasmer_engine.go:215]: nice typo to fix

**Review by ethanfrey — APPROVED** [review]
Good stuff. Will merge.

Left some minor comments that may be worth addressing in the future

**Pull request merged**